packages/@ama-sdk/core/src/plugins/additional-params/additional-params-sync.request.ts

```
@@ -38,7 +44,11 @@ export class AdditionalParamsSyncRequest implements RequestPlugin {
         const body = this.additionalParams.body && isStringOrUndefined(data.body) ? this.additionalParams.body(data.body) : undefined;
 
         if (queryParams) {
-          data.queryParams = { ...data.queryParams, ...queryParams };
+          if (data.paramSerializationOptions?.enableParameterSerialization && !isParamValueRecord(queryParams)) {
+            data.queryParams = { ...data.queryParams, ...serializeRequestPluginQueryParams(queryParams) };
+          } else if (isParamValueRecord(queryParams)) {
```

should we log/throw an error in case both `data.paramSerializationOptions?.enableParameterSerialization` and `isParamValueRecord(queryParams)` are false?
I assume here the params are silently ignored

---

I could throw an error yes if you find this helpful, I was not sure when to add error logs or not

---

updated